substrate/frame/aura/src/default_weights.rs

```diff
@@ -0,0 +1,54 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Default weights for the AURA Pallet.
+//! This file was not auto-generated.
+
+use frame_support::weights::{
+	constants::{RocksDbWeight as DbWeight, WEIGHT_REF_TIME_PER_MICROS, WEIGHT_REF_TIME_PER_NANOS},
+	Weight,
+};
+
+/// Default `WeightInfo` implementation generic over the max number of validator's nominators (`N`).
+pub struct SubstrateWeight<const N: u32>;
```

Comment:
Here I tried to experiment and detach a bit the consensus from the staking.

That is, instead of adding the `MaxNominators` directly in the `Config`, I decided to provide the (const) value as part of the weight type we provide.

IMO this makes sense as:
1. relax the dependency to the staking model which the user may don't want to use.
2. the `MaxNominators` value is only used here, to compute the weight

Maybe I can also provide the `()` implementation where we set the max nominators to `0`.

If the idea is good, I'm going to do the same for the other consensus protocols (in a follow-up PR)

cc @andresilva